### PR TITLE
Update the Rubocop config

### PR DESCRIPTION
**Pull request by lslezak**
to make it work with Rubocop 0.40.0

### CODE REVIEW

File: ytools/y2tool/rubocop_yast_style.yml [hunk 1]
```diff
@@ -199,3 +202,46 @@ Style/AsciiComments:
...
+  Enabled: false
+
+# Use the backward compatible default
+Style/MultilineMethodCallIndentation:
+  EnforcedStyle: indented
+
+# Do not check the module length, the YCP converted includes
+# are usually too long.
+Metrics/ModuleLength:
+  Enabled: false
```
Comment (jreidinger) [inline ytools/y2tool/rubocop_yast_style.yml]: this is wrong, we do not do it even for class length or method lenght...It is intentionally undefined, so if you generate rubocop config it adds current value and it try to check that it is not increased.

File: ytools/y2tool/rubocop_yast_style.yml [hunk 2]
```diff
@@ -199,3 +202,46 @@ Style/AsciiComments:
...
+
+# Disable the "foo.size == 0" check, Yast::Term does not provide #empty?
```
Comment (jreidinger) [inline ytools/y2tool/rubocop_yast_style.yml]: ugh, what? Yast::Term have `#empty?`! see https://github.com/yast/yast-ruby-bindings/blob/master/src/ruby/yast/term.rb#L37

File: ytools/y2tool/rubocop_yast_style.yml [hunk 3]
```diff
@@ -199,3 +202,46 @@ Style/AsciiComments:
...
+Style/ZeroLengthPredicate:
+  Enabled: false
+
+# Use the backward compatible default
```
Comment (jreidinger) [inline ytools/y2tool/rubocop_yast_style.yml]: I am not sure if we want to make use backward compatible default or have some discussion which style we prefer and then let rubocop auto reformat it.

File: ytools/y2tool/rubocop_yast_style.yml (hunk 2 above)
Comment (lslezak) [inline ytools/y2tool/rubocop_yast_style.yml]: Um, I saw [this comment](https://github.com/yast/yast-yast2/blob/8eb9214265fdf8afcae1531c5a0309799028d79d/library/packages/src/modules/PackagesUI.rb#L317) and disabled it without verifying that...

I'll enable it.

Comment (jreidinger) [inline ytools/y2tool/rubocop_yast_style.yml]: It is not there from beginning, but I add it as I found it quite useful and there is even more methods, that can help operating with terms :wink:

**Comment by jreidinger** [comment]
now it in general looks good, but I think that it deserve some discussion on mailing list about backward compatible ones, so we can decide which way we prefer and maybe we can force now recommended value?

In general I think we should follow upstream guide unless we have good reason to change it. And backward compatibility in style guide is not good reason for me as it can be auto-corrected.

**Comment by lslezak** [comment]
OK, makes sense, initially I wanted to change the code as little as possible, but the argument about the upstream compatibility is really strong. I'll update it...

File: ytools/y2tool/rubocop_yast_style.yml [hunk 4]
```diff
@@ -199,3 +201,37 @@ Style/AsciiComments:
...
+  EnforcedStyle: consistent
+
+# Use the backward compatible default
+Lint/EndAlignment:
+  AlignWith: start_of_line
+
+#  Allow block.call:
+#
+#  def foo(&block)
+#    var = block,call
```
Comment (mvidner) [inline ytools/y2tool/rubocop_yast_style.yml]: dot instead of comma

**Comment by jreidinger** [comment]
LGTM

**Comment by jreidinger** [comment]
LGTM

**Pull request merged**